firmware/raspi/pieces.py

```diff
@@ -0,0 +1,228 @@
+"""Module containing helper GUI code for gui.py.
+
+Based on https://github.com/saavedra29/chess_tk"""
+
+from shutil import which
+import sys
+from tracemalloc import start
+
+SHORT_NAME = {
+    "R": "Rook",
+    "N": "Knight",
+    "B": "Bishop",
+    "Q": "Queen",
+    "K": "King",
+    "P": "Pawn"
+}
+
+leftBlackMoved, rightBlackMoved, leftWhiteMoved, rightWhiteMoved = False, False, False, False
+leftBlackCheck, rightBlackCheck, leftWhiteCheck, rightWhiteCheck = False, False, False, False
+whiteKingHome, blackKingHome = True, True
+
+def create_piece(piece, color="white"):
+    """Takes a piece name or shortname and returns the corresponding piece
+    instance
+    """
+    if piece in (None, " "):
+        return
+    if len(piece) == 1:
+        if piece.isupper():
+            color = "white"
+        else:
+            color = "black"
+        piece = SHORT_NAME[piece.upper()]
+    module = sys.modules[__name__]
+    return module.__dict__[piece](color)
+
+class Piece(object):
+    """Chess piece class containing color and shortname.
+    """
+    def __init__(self, color):
+        if color == "black":
+            self.shortname = self.shortname.lower()
+        elif color == "white":
+            self.shortname = self.shortname.upper()
+        self.color = color
+
+    def place(self, chessboard):
+        """Keep a reference to the board.
+        """
+        self.board = chessboard
+
+    def moves_available(self, pos, orthogonal, diagonal, distance):
+        """Creates array of all valid moves
+        """
+        board = self.board
+        allowed_moves = []
+        orth = ((-1, 0), (0, -1), (0, 1), (1, 0))
+        diag = ((-1, -1), (-1, 1), (1, -1), (1, 1))
+        piece = self
+        beginningpos = board.num_notation(pos.upper())
+        if orthogonal and diagonal:
+            directions = diag + orth
+        elif diagonal:
+            directions = diag
+        elif orthogonal:
+            directions = orth
+        for x, y in directions:
+            collision = False
+            for step in range(1, distance + 1):
+                if collision: break
+                dest = beginningpos[0] + step * x, beginningpos[1] + step * y
+                if self.board.alpha_notation(dest) not in board.occupied(
+                        "white") + board.occupied("black"):
+                    allowed_moves.append(dest)
+                elif self.board.alpha_notation(dest) in board.occupied(
+                        piece.color):
+                    collision = True
+                else:
+                    allowed_moves.append(dest)
+                    collision = True
+        allowed_moves = filter(board.is_on_board, allowed_moves)
+        return map(board.alpha_notation, allowed_moves)
+
+class King(Piece):
+    """King chess piece.
+    """
+    shortname = "k"
+    canCastle = True
```

Comment:
Please update all camel case variable names to be snake case instead: e.g. `can_castle` instead of `canCastle`.